ambari/datadog_checks/ambari/ambari.py

```diff
@@ -0,0 +1,198 @@
+
+        if components_response is None:
+            self.log.warning("No components found for service {}.".format(service))
+            return
+
+        for component in components_response.get('items'):
+            component_name = component.get('ServiceComponentInfo').get('component_name')
+
+            if component_name not in component_whitelist:
+                self.log.warning('{} not found in {}:{}'.format(component_name, cluster, service))
+                continue
+            if component.get(METRICS_FIELD) is None:
+                self.log.warning(
+                    "No metrics found for component {} for service {}"
+                        .format(component_name, service)
+                )
+                continue
+
+            for header in metric_whitelist:
+                if header not in component.get(METRICS_FIELD):
+                    self.log.warning(
+                        "No {} metrics found for component {} for service {}"
+                            .format(header, component_name, service)
+                    )
+                    continue
+
+                metrics = self.flatten_service_metrics(component.get(METRICS_FIELD)[header], header)
+                component_tag = COMPONENT_TAG + component_name.lower()
+                for metric_name, value in iteritems(metrics):
+                    metric_tags = base_tags + [component_tag]
+                    if isinstance(value, float):
+                        self._submit_gauge(metric_name, value, metric_tags)
+                    else:
+                        self.log.warning("Expected a float for {}, received {}".format(metric_name, value))
+
+    def _get_hosts_info(self, base_url, authentication, cluster):
+        hosts_endpoint = common.HOST_METRICS_URL.format(
+            base_url=base_url,
+            cluster_name=cluster
+        )
+        resp = self._make_request(hosts_endpoint, authentication)
+
+        return resp.get('items')
+
+    def _get_service_checks_info(self, base_url, authentication, cluster, service, service_tags):
+        service_check_endpoint = common.create_endpoint(base_url, cluster, service, SERVICE_INFO_QUERY)
+        service_info = []
+        service_resp = self._make_request(service_check_endpoint, authentication)
+        if service_resp is None:
+            service_info.append({'state': self.CRITICAL, 'tags': service_tags})
+            self.log.warning("No response received for service {}".format(service))
+        else:
+            state = service_resp.get('ServiceInfo').get('state')
+            service_info.append({'state': common.STATUS[state], 'tags': service_tags})
+        return service_info
+
+    def _make_request(self, url, auth):
+        try:
+            resp = self.http.get(url,
+                                 auth=HTTPBasicAuth(auth.get('username'), auth.get('password')),
+                                 verify=False)  # In case Ambari is under uncertified https
```

Comment:
This is handled by the wrapper as well. You need to add the proper options to the config file.